### PR TITLE
Nix updates

### PR DESCRIPTION
**Pull request by muscaln**
Bumps flake.lock and adds support for Qt6.

### CODE REVIEW

**Comment by Scrumplex** [comment]
Current codebase won't build with Qt 6. There is some progress for Qt6 support in #575

**Comment by muscaln** [comment]
`nix flake check` fails because Qt6 is not landed in `nixpkgs-unstable` branch. Its available in `nixos-unstable`.

```
❯ nix flake check
error: attribute 'qt6Packages' missing

       at /nix/store/1z1517kwbfy0b6pdxq786cy9il3vdbf8-source/flake.nix:27:22:

           26|         polymc = pkgs.${system}.libsForQt5.callPackage ./nix { inherit version self libnbtplusplus; };
           27|         polymc-qt6 = pkgs.${system}.qt6Packages.callPackage ./nix { inherit version self libnbtplusplus; };
             |                      ^
           28|       });
```

File: nix/default.nix [hunk 1]
```diff
@@ -35,38 +35,36 @@ let
...
+    mkdir source/libraries/libnbtplusplus
     cp -a ${libnbtplusplus}/* source/libraries/libnbtplusplus
-    cp -a ${quazip}/* source/libraries/quazip
-    chmod a+r+w source/libraries/{libnbtplusplus,quazip}/*
+    chmod a+r+w source/libraries/libnbtplusplus/*
   '';
 
   cmakeFlags = [
     "-GNinja"
     "-DLauncher_PORTABLE=OFF"
```
Comment (flowln) [inline nix/default.nix]: The flag is not used anymore, so you can remove it :)
Also, you may be interested in setting the `ENABLE_LTO` one

File: nix/default.nix [hunk 2]
```diff
@@ -35,38 +35,36 @@ let
...
-  ];
+    "-DLauncher_QT_VERSION_MAJOR=${lib.versions.major qtbase.version}"
+  ] ++ lib.optionals (msaClientID != "") [ "-DLauncher_MSA_CLIENT_ID=${msaClientID}" ];
```
Comment (flowln) [inline nix/default.nix:62]: Is this needed? PolyMC has an in-launcher field to change the MSA Client ID already

**Review by flowln — COMMENTED** [review]
I don't know much about Nix, so sorry if any of my comments is *obviously incorrect* x.x

File: nix/default.nix (hunk 2 above)
Comment (muscaln) [reply nix/default.nix:62]: I think we can keep it since it's optional.

**Review by timoreo22 — APPROVED** [review]

**Review by flowln — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Comment by Scrumplex** [comment]
Thanks!

**Comment by ncfavier** [comment]
https://github.com/PolyMC/PolyMC/pull/653#issuecomment-1139997288 causes the import into nixos-search to fail. Needs another lockfile update.

**Comment by muscaln** [comment]
> [#653 (comment)](https://github.com/PolyMC/PolyMC/pull/653#issuecomment-1139997288) causes the import into nixos-search to fail. Needs another lockfile update.



#762

**Pull request merged**